sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -110,12 +119,31 @@ def test_kernel_pca_linear_kernel():
         np.abs(PCA(4).fit(X_fit).transform(X_pred)))
 
 
+def test_kernel_pca_linear_kernel2():
+    """ Tests that kPCA with a linear kernel is equivalent to PCA, for all
+    solvers"""
+    rng = np.random.RandomState(0)
+    X_fit = rng.random_sample((6, 10))
+    X_pred = rng.random_sample((2, 10))
+
+    # for a linear kernel, kernel PCA should find the same projection as PCA
+    # modulo the sign (direction)
+    for solver in ("auto", "dense", "arpack", "randomized"):
+        assert_array_almost_equal(
+            np.abs(KernelPCA(4, eigen_solver=solver).fit(X_fit)
+                   .transform(X_pred)),
+            np.abs(PCA(4, svd_solver=solver if solver != "dense" else "full")
```

Comment:
We could probably harmonize the "full" vs "dense" values in the two classes (with some deprecation cycle to ensure backward compat) but this should probably be done in a later PR.

Reply:
agreed. I can do it here if you wish, by simply supporting both in the code and documenting only the new one in docstring. Let me know if you wish me to do it.

Reply:
ok created #16917